lib/class/art.class.php

```diff
@@ -948,8 +948,17 @@ public function generate_thumb($image, $size, $mime)
      * @param string $type
      * @return string
      */
-    public static function get_from_source($data, $type = 'album')
```

Comment:
why did you remove the default value for parameter 'type'?

Reply:
did a search on it and don't see any calls with a null type, but it's still a fallback i would keep

Reply:
My problem was that the function always used the default value 'album', although I called the function with the parameter value 'song'. My parameter was ignored, it always was overwritten by 'album'. I had a long debugging session until I realized this.

But to be honest I'm absolutely new to PHP - maybe the reason was different?

Reply:
i'll check it out, it's possible that something is being sent null

Reply:
Well here is the [php authority](https://www.php.net/manual/en/).  It's always good to have a default value: no one knows how it migt be accessed in the future.

Reply:
There's code below to set album/song based on the config. i'm not sure when this would ever fire but there must be something not format or set before the call somewhere outside this
```
public static function get_from_source($data, $type)
    {
        if (!isset($type)) {
            $type = (AmpConfig::get('show_song_art')) ? 'song' : 'album';
        }
```